core/src/packets/arp.rs

```diff
@@ -0,0 +1,658 @@
+/*
+* Copyright 2019 Comcast Cable Communications Management, LLC
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* SPDX-License-Identifier: Apache-2.0
+*/
+
+//! Address Resolution Protocol.
+
+use crate::net::MacAddr;
+use crate::packets::types::u16be;
+use crate::packets::{EtherTypes, Ethernet, Internal, Packet, ParseError};
+use crate::{ensure, SizeOf};
+use failure::Fallible;
+use std::fmt;
+use std::net::Ipv4Addr;
+use std::ptr::NonNull;
+
+/// Address Resolution Protocol packet based on [IETF RFC 826].
+///
+/// ```
+///  0                   1                   2                   3
+///  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+/// |         Hardware Type         |         Protocol Type         |
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+/// |    H Length   |    P Length   |         Operation Code        |
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+/// |                   Source Hardware Address                     
```

Comment:
isn't it sender not source typically (in standard nomenclature?)?

Reply:
I must be asleep. not sure where source came from. everywhere you look it's sender.